### PR TITLE
Update components to the more recent versions.

### PR DESCRIPTION
**Pull request by mwringe**
Fixes issue when running with newer versions of OpenShift

### CODE REVIEW

File: cassandra/Dockerfile [hunk 1]
```diff
@@ -39,24 +38,28 @@ ENV CASSANDRA_VERSION="2.2.1" \
 # Become the root user to be able to install and setup Cassandra under /opt
 USER root
 
+#
+RUN yum install -y -q bind-utils && \
+    yum clean all
+
+COPY apache-cassandra-2.2.4-bin.tar.gz.md5 \
```
Comment (jcantrill) [inline cassandra/Dockerfile]: Are we only able to get this md5 by checking it in locally? If there was a way to parameterize, we could propagate the CASSANDRA_VERSION throughout the Dockerfile

File: cassandra/Dockerfile [hunk 2]
```diff
@@ -39,24 +38,28 @@ ENV CASSANDRA_VERSION="2.2.1" \
...
+     /tmp
+
 # Copy the Cassandra binary to the /opt directory
 RUN cd /opt; \
     curl -LO https://archive.apache.org/dist/cassandra/$CASSANDRA_VERSION/apache-cassandra-$CASSANDRA_VERSION-bin.tar.gz && \
+    if [ "$(md5sum apache-cassandra-2.2.4-bin.tar.gz | awk '{print $1}')" != "$(cat /tmp/apache-cassandra-2.2.4-bin.tar.gz.md5)" ]; then exit 1; fi && \
```
Comment (jcantrill) [inline cassandra/Dockerfile]: Maybe we should add a print statement explicitly identifying why we are exiting early

File: cassandra/gather-seeds.sh [hunk 3]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+#How many retries before giving up
+RETRIES=30
```
Comment (jcantrill) [inline cassandra/gather-seeds.sh]: Any need to configure this by passing in a value?

File: deployer/templates/hawkular-cassandra-node-pv.yaml [hunk 4]
```diff
@@ -77,7 +77,7 @@ objects:
             containerPort: 7001
           command:
           - "/opt/apache-cassandra/bin/cassandra-docker.sh"
-          - "--seed_provider_classname=org.hawkular.openshift.cassandra.OpenshiftSeedProvider"
+          - "--seed_provider_classname=org.apache.cassandra.locator.SimpleSeedProvider"
```
Comment (jcantrill) [inline deployer/templates/hawkular-cassandra-node-pv.yaml]: Any need to be able to pass this in as an ENV var?

**Comment by jcantrill** [comment]
Other then my few comments, LGTM. squash and merge

File: deployer/templates/hawkular-cassandra-node-pv.yaml (hunk 4 above)
Comment (mwringe) [inline deployer/templates/hawkular-cassandra-node-pv.yaml]: In the previous version we were using a custom seed provider. This meant for the one running in OpenShift we would use the custom seed provider, while if you wanted to run just the plain docker image (without OpenShift) you could use the default one. If you wanted to use a custom seed provider you could also extend our base image and change the command option to point to the one you want to use instead.

We can probably remove the option in the configuration and just have it use the default, but I don't think we would want to remove the functionality.

File: cassandra/gather-seeds.sh (hunk 3 above)
Comment (mwringe) [inline cassandra/gather-seeds.sh]: We could make it configurable, but I don't think it adds much value (and we already have a lot of configuration options as is).

If the retries are used up, the container will exit under an error condition and a new container will be started by the rc. The RETRIES values just determines how many times we try before exiting and restarting again.

File: cassandra/Dockerfile (hunk 1 above)
Comment (mwringe) [inline cassandra/Dockerfile]: Its possible to download the md5 sum from the server (its slightly more secure to have it locally installed though).

But in the very least we should at least go through and change the 2.2.4 instances to ${CASSANDRA_VERSION} as that will make things easier

**Comment by mwringe** [comment]
Updated based on comments and squashed.

**Pull request merged**